### ISSUE TITLE
EnumerateEnvironmentVariables test failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#16481]
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/3862/consoleText
```
     System.Tests.GetEnvironmentVariable.EnumerateEnvironmentVariables(target: Process) [FAIL]
        Assert.Equal() Failure
        Expected: false
        Actual:   (null)
        Stack Trace:
           D:\j\workspace\windows_nt_de---4526f5ff\src\System.Runtime.Extensions\tests\System\Environment.GetEnvironmentVariable.cs(199,0): at System.Tests.GetEnvironmentVariable.EnumerateEnvironmentVariables(EnvironmentVariableTarget target)
```
cc: @JeremyKuhne

**Comment by JeremyKuhne** [comment]
Urgh, another test timing problem. There is no guarantee the value we iterate will still exist (or not be changed in some other way) by the time we look it up again. In this case I'm 99% sure that we're colliding with the other test classes.

I'll change this to set a specific variable and only compare that one.

On a related note it is unfortunate that Assert.Equal() doesn't allow adding additional context.

**Comment by karelz** [comment]
@JeremyKuhne I marked the bug with labels and milestone on your behalf.

**Comment by stephentoub** [comment]
https://ci.dot.net/job/dotnet_corefx/job/master/job/portablelinux_release_prtest/1204/consoleText

**Comment by jkotas** [comment]
https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu14.04_release_prtest/3946/

**Comment by sepidehMS** [comment]
https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu14.04_debug_prtest/3937/

**Comment by JeremyKuhne** [comment]
Fixed with #16546
